site/content/features/virtual-threads.md

```diff
@@ -0,0 +1,433 @@
+
+{{< sandbox version=java21 preview="true" mainclass="VirtualThreadDemo" >}}{{< sandboxsource "VirtualThreadDemo.java" >}}
+import java.util.concurrent.*;
+import java.net.*;
+import java.net.http.*;
+
+public class VirtualThreadDemo {
+   public static void main(String[] args) throws InterruptedException, ExecutionException {
+      ExecutorService service = Executors.newVirtualThreadPerTaskExecutor();
+      Future<String> f1 = service.submit(() -> get("https://horstmann.com/random/adjective"));
+      Future<String> f2 = service.submit(() -> get("https://horstmann.com/random/noun"));
+      String result = f1.get() + " " + f2.get();
+      System.out.println(result);
+      service.close();
+   }
+
+   private static HttpClient client = HttpClient.newHttpClient();
+
+   public static String get(String url) {
+      try {
+         var request = HttpRequest.newBuilder().uri(new URI(url)).GET().build();
+         return client.send(request, HttpResponse.BodyHandlers.ofString()).body();
+      } catch (Exception ex) {
+         throw new RuntimeException(ex);
+      }
+   }   
+}
+
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+If you have a list of tasks with the same result type, you can use the `invokeAll` method and then call `get` on each `Future`:
+
+```
+List<Callable<T>> callables = ...;
+List<T> results = new ArrayList<>();
+for (Future<T> f : service.invokeAll(callables))
+  results.add(f.get());
+```
+
+Again, a more concrete sandbox:
+
+{{< sandbox version=java21 preview="true" mainclass="VirtualThreadDemo" >}}{{< sandboxsource "VirtualThreadDemo.java" >}}
+import java.util.*;
+import java.util.concurrent.*;
+import java.net.*;
+import java.net.http.*;
+
+public class VirtualThreadDemo {
+   public static void main(String[] args) throws InterruptedException, ExecutionException {
+      ExecutorService service = Executors.newVirtualThreadPerTaskExecutor();
+      List<Callable<String>> callables = new ArrayList<>();
+      final int ADJECTIVES = 4;
+      for (int i = 1; i <= ADJECTIVES; i++)
+         callables.add(() -> get("https://horstmann.com/random/adjective"));
+      callables.add(() -> get("https://horstmann.com/random/noun"));
+      List<String> results = new ArrayList<>();
+      for (Future<String> f : service.invokeAll(callables))
+         results.add(f.get());
+      System.out.println(String.join(" ", results));
+      service.close();
+   }
+
+   private static HttpClient client = HttpClient.newHttpClient();
+
+   public static String get(String url) {
+      try {
+         var request = HttpRequest.newBuilder().uri(new URI(url)).GET().build();
+         return client.send(request, HttpResponse.BodyHandlers.ofString()).body();
+      } catch (Exception ex) {
+         var rex = new RuntimeException();
+         rex.initCause(ex);
+         throw rex;
+      }
+   }   
```

Comment:
Due to usage of callable exception handler is not required:

```suggestion
   public static String get(String url) throws Exception {
      var request = HttpRequest.newBuilder().uri(new URI(url)).GET().build();
      return client.send(request, HttpResponse.BodyHandlers.ofString()).body();
   }  
```

Reply:
Changed to `public static String get(String url) throws InterruptedException, IOException, URISyntaxException`